File: Daily Test/C++/Generate Number From Matrix 24.5.2025.cpp
#include <iostream>
using namespace std;
int main() {
    int matrix[9][9];
    string N;
    for(int i = 0; i < 9; i++)
        for(int j = 0; j < 9; j++)
            cin >> matrix[i][j];
    cin >> N;
    int result = 0;
    for(int j = 0; j < 9; j++) {
        int digits = N[j] - '0', num = 0;
        for(int i = 0; i < digits; i++)
            num = num * 10 + matrix[i][j];
        result += num;
    }
    cout << result << endl;
    return 0;
}

File: Daily Test/Java/Generate Number From Matrix 24.5.2025.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[][] matrix = new int[9][9];
        for(int i = 0; i < 9; i++)
            for(int j = 0; j < 9; j++)
                matrix[i][j] = sc.nextInt();
        String N = sc.next();
        int result = 0;
        for(int j = 0; j < 9; j++) {
            int digits = N.charAt(j) - '0';
            StringBuilder numStr = new StringBuilder();
            for(int i = 0; i < digits; i++)
                numStr.append(matrix[i][j]);
            result += Integer.parseInt(numStr.toString());
        }
        System.out.println(result);
    }
}